core/src/test/scala/unit/kafka/tools/MirrorMakerTest.scala

```diff
@@ -17,12 +17,82 @@
 
 package kafka.tools
 
-import kafka.consumer.BaseConsumerRecord
+import java.util
+import java.util.Properties
+
+import kafka.consumer.{BaseConsumerRecord}
+import kafka.server.{KafkaConfig, KafkaServer}
+import kafka.tools.MirrorMaker.MirrorMakerNewConsumer
+import kafka.utils.{CoreUtils, TestUtils}
+import kafka.zk.ZooKeeperTestHarness
+import org.apache.kafka.clients.consumer.{ConsumerConfig, KafkaConsumer}
+import org.apache.kafka.clients.producer.{KafkaProducer, ProducerRecord}
 import org.apache.kafka.common.record.{Record, TimestampType}
 import org.junit.Assert._
-import org.junit.Test
+import org.junit.{After, Before, Test}
+
+class MirrorMakerTest extends ZooKeeperTestHarness {
+
+  private var server1: KafkaServer = null
+
+  @Before
+  override def setUp() {
+    super.setUp()
+    val props1 = TestUtils.createBrokerConfig(0, zkConnect, false)
+    val config1 = KafkaConfig.fromProps(props1)
+    server1 = TestUtils.createServer(config1)
 
-class MirrorMakerTest {
+    val topic = "new-topic"
+    TestUtils.createTopic(zkUtils, topic, numPartitions = 1, replicationFactor = 1, servers = List(server1))
+  }
+
+  @After
+  override def tearDown() {
+    server1.shutdown
+    CoreUtils.delete(server1.config.logDirs)
+    super.tearDown()
+  }
+
+  @Test
+  def testWhitelistTopic() {
+    val topic = "new-topic"
+    val msg = "test message"
+    val brokerList = TestUtils.getBrokerListStrFromServers(Seq(server1))
+
+    // Create a test producer to delivery a message
+    val props = new Properties()
+    props.put("bootstrap.servers", brokerList)
+    props.put("key.serializer", "org.apache.kafka.common.serialization.ByteArraySerializer")
+    props.put("value.serializer", "org.apache.kafka.common.serialization.ByteArraySerializer")
+    val producer1 = new KafkaProducer[Array[Byte], Array[Byte]](props)
+    producer1.send(new ProducerRecord[Array[Byte], Array[Byte]](topic, msg.getBytes()))
+    producer1.flush() // Explicitly invoke flush method to make effect immediately
+    producer1.close() // Close the producer
+
+    // Create a MirrorMaker consumer
+    val config = new util.HashMap[String, AnyRef]
```

Comment:
Is there a reason why we use a `Properties` instance for the producer and a `HashMap` for the consumer? It would be good to be consistent (either choice is fine).